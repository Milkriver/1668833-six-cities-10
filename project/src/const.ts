import { City } from './types/offer';

export enum AppRoute {
  Main = '/',
  Login = '/login',
  Favorites = '/favorites',
  Room = '/offer/:id',
  PageError = '*'
}

export enum AuthorizationStatus {
  Auth = 'AUTH',
  NoAuth = 'NO_AUTH',
  Unknown = 'UNKNOWN',
}

export const isCheckedAuth = (authorizationStatus: AuthorizationStatus): boolean =>
  authorizationStatus === AuthorizationStatus.Unknown;

export enum APIRoute {
  Hotels = '/hotels',
  Hotel = '/hotels/:id',
  HotelsNearby = '/hotels/:id/nearby',
  Favorites = '/favorite',
  Comments = '/comments/:id',
  Comment = '/comments/:id',
  Login = '/login',
  Logout = '/logout',
}

export const URL_MARKER_DEFAULT = 'img/pin.svg';
export const URL_MARKER_CURRENT = 'img/pin-active.svg';

export const locations: Record<string, City> = {
  Amsterdam: {
    name: 'Amsterdam',
    location: {
      latitude: 52.37454,
      longitude: 4.897976,
      zoom: 13
    }
  },
  Paris: {
    name: 'Paris',
    location: {
      latitude: 48.8534,
      longitude: 2.3488,
      zoom: 13
    }
  },
  Cologne: {
    name: 'Cologne',
    location: {
      latitude: 45.5774872,
      longitude: 9.939068899999999,
      zoom: 13
    }
  },
  Brussels: {
    name: 'Brussels',
    location: {
      latitude: 50.8504,
      longitude: 4.34878,
      zoom: 13
    }
  },
  Hamburg: {
    name: 'Hamburg',
    location: {
      latitude: 53.5753,
      longitude: 10.0153,
      zoom: 13
    }
  },
  Dusseldorf: {
    name: 'Dusseldorf',
    location: {
      latitude: 51.2217,
      longitude: 6.77616,
      zoom: 13
    }
  },
};

export const sortOptionList = {
  popular: 'Popular',
  risingPrice: 'Price: low to high',
  decliningPrice: 'Price: high to low',
  topRating: 'Top rated first',
};
