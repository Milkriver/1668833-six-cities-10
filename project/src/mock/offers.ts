import { Offers } from '../types/offer';

export const offers:Offers = [
  {
    id: 1,
    image: ['img/room.jpg', 'img/apartment-01.jpg', 'img/apartment-02.jpg', 'img/apartment-03.jpg', 'img/studio-01.jpg', 'img/apartment-01.jpg'],
    premium: true,
    bookmark: false,
    name: 'Beautiful & luxurious apartment at great location',
    type: 'Apartment',
    price: 120,
    rating: 4.8,
    bedrooms: 3,
    guests: 4,
    options: ['Wi-Fi', 'Washing machine', 'Towels', 'Heating', 'Coffee machine', 'Baby seat', 'Kitchen', 'Dishwasher', 'Cabel TV', 'Fridge'],
    host: {
      image: 'img/avatar-angelina.jpg',
      name: 'Angelina',
      status: 'Pro',
    },
    description: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.The building is green and from 18th century. An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.',
    reviews: [
      {
        avatar: 'img/avatar-max.jpg',
        name: 'Max',
        text: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.The building is green and from 18th century.',
        date: '2019-04-24',
        rating: 4,
      }
    ],
  },
  {
    id: 2,
    image: ['img/room.jpg', 'img/apartment-01.jpg', 'img/apartment-02.jpg', 'img/apartment-03.jpg', 'img/apartment-01.jpg'],
    premium: false,
    bookmark: true,
    name: 'Wood and stone place',
    type: 'Private room',
    price: 80,
    rating: 4,
    bedrooms: 3,
    guests: 4,
    options: ['Wi-Fi', 'Heating', 'Coffee machine', 'Kitchen', 'Dishwasher', 'Cabel TV', 'Fridge'],
    host: {
      image: 'img/avatar-angelina.jpg',
      name: 'Angelina',
      status: 'Pro',
    },
    description: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.The building is green and from 18th century. An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.',
    reviews: [
      {
        avatar: 'img/avatar-max.jpg',
        name: 'Max',
        text: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.The building is green and from 18th century.',
        date: '2019-04-24',
        rating: 4,
      }
    ],
  },
  {
    id: 3,
    image: ['img/apartment-02.jpg', 'img/apartment-03.jpg', 'img/apartment-01.jpg'],
    premium: false,
    bookmark: false,
    name: 'Canal View Prinsengracht',
    type: 'Apartment',
    price: 132,
    rating: 4,
    bedrooms: 3,
    guests: 4,
    options: ['Wi-Fi', 'Heating', 'Coffee machine', 'Kitchen', 'Dishwasher', 'Cabel TV', 'Fridge'],
    host: {
      image: 'img/avatar-angelina.jpg',
      name: 'Angelina',
      status: 'Pro',
    },
    description: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.The building is green and from 18th century. An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.',
    reviews: [
      {
        avatar: 'img/avatar-max.jpg',
        name: 'Max',
        text: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.The building is green and from 18th century.',
        date: '2019-04-24',
        rating: 4,
      }
    ],
  },
  {
    id: 4,
    image: ['img/apartment-03.jpg', 'img/apartment-01.jpg'],
    premium: false,
    bookmark: false,
    name: 'Nice, cozy, warm big bed apartment',
    type: 'Apartment',
    price: 125,
    rating: 4,
    bedrooms: 2,
    guests: 2,
    options: ['Wi-Fi', 'Coffee machine', 'Kitchen', 'Dishwasher', 'Cabel TV', 'Fridge'],
    host: {
      image: 'img/avatar-angelina.jpg',
      name: 'Helen',
      status: 'Pro',
    },
    description: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.The building is green and from 18th century. An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.',
    reviews: [
      {
        avatar: 'img/avatar-max.jpg',
        name: 'Max',
        text: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.The building is green and from 18th century.',
        date: '2019-04-24',
        rating: 4,
      }
    ],
  },
];
