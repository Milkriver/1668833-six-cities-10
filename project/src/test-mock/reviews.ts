export const mockReviews = [
  {
    comment: 'The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.',
    date: '2022-06-23T12:25:36.939Z',
    id: 1,
    rating: 3,
    user: {
      avatarUrl: 'https://10.react.pages.academy/static/avatar/3.jpg',
      id: 12,
      isPro: true,
      name: 'Isaac'
    }
  },
  {
    comment: 'I stayed here for one night and it was an unpleasant experience.',
    date: '2022-06-23T12:25:36.939Z',
    id: 2,
    rating: 5,
    user: {
      avatarUrl: 'https://10.react.pages.academy/static/avatar/2.jpg',
      id: 11,
      isPro: false,
      name: 'Jack'
    }
  }
];
