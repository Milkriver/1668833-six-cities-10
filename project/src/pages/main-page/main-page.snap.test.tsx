import {render} from '@testing-library/react';
import MainPage from './main-page';
import { configureMockStore } from '@jedmao/redux-mock-store';
import { Provider } from 'react-redux';
import { AuthorizationStatus } from '../../const';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import HistoryRouter from '../../components/history-router/history-router';
import { mockOffers } from '../../test-mock/offers';


describe('Component: MainPage', () => {
  test('should render correctly', () => {
    const middlewares = [thunk];
    const mockStore = configureMockStore(middlewares);

    const store = mockStore({
      USER: {
        authorizationStatus: AuthorizationStatus.Auth,
        userEmail: 'ma@ya.ru',
      },
      DATA: {
        favoriteOffers: [],
        city: mockOffers[0].city,
        offers: mockOffers
      }
    });

    const history = createMemoryHistory();

    const {container} = render(
      <Provider store={store}>
        <HistoryRouter history={history}>
          <MainPage />
        </HistoryRouter>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
