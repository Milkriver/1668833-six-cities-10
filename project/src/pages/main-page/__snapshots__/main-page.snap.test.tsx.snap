// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: MainPage should render correctly 1`] = `
<div>
  <div
    class="page page--gray page--main"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link header__logo-link--active"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/favorites"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    class="header__user-name user__name"
                  >
                    ma@ya.ru
                  </span>
                  <span
                    class="header__favorite-count"
                  >
                    0
                  </span>
                </a>
              </li>
              <li
                class="header__nav-item"
              >
                <a
                  class="header__nav-link"
                  href="/"
                >
                  <span
                    class="header__signout"
                  >
                    Sign out
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--index"
    >
      <h1
        class="visually-hidden"
      >
        Cities
      </h1>
      <div
        class="tabs"
      >
        <section
          class="locations container"
        >
          <ul
            class="locations__list tabs__list"
          >
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="#"
              >
                <span>
                  Paris
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="#"
              >
                <span>
                  Cologne
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="#"
              >
                <span>
                  Brussels
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="#"
              >
                <span>
                  Amsterdam
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="#"
              >
                <span>
                  Hamburg
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item tabs__item--active"
                href="#"
              >
                <span>
                  Dusseldorf
                </span>
              </a>
            </li>
          </ul>
        </section>
      </div>
      <div
        class="cities"
      >
        <div
          class="cities__places-container container"
        >
          <section
            class="cities__places places"
          >
            <h2
              class="visually-hidden"
            >
              Places
            </h2>
            <b
              class="places__found"
            >
              1
               places to stay in 
              Dusseldorf
            </b>
            <form
              action="#"
              class="places__sorting"
              method="get"
            >
              <span
                class="places__sorting-caption"
              >
                Sort by
              </span>
              <span
                class="places__sorting-type"
              >
                <svg
                  class="places__sorting-arrow"
                  height="4"
                  width="7"
                >
                  <use
                    xlink:href="#icon-arrow-select"
                  />
                </svg>
              </span>
            </form>
            <div
              class="cities__places-list places__list tabs__content"
            >
              <article
                class="cities__card place-card"
                data-testid="offer-card__box"
              >
                <div
                  class="cities__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/1"
                  >
                    <img
                      alt="Penthouse, 4-5 rooms + 5 balconies"
                      class="place-card__image"
                      height="200"
                      src="https://10.react.pages.academy/static/hotel/18.jpg"
                      width="260"
                    />
                  </a>
                </div>
                <div
                  class="cities__card-info place-card__info"
                >
                  <div
                    class="place-card__price-wrapper"
                  >
                    <div
                      class="place-card__price"
                    >
                      <b
                        class="place-card__price-value"
                      >
                        €
                        393
                      </b>
                      <span
                        class="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      class="place-card__bookmark-button button false"
                      data-testid="offer-card__favorite"
                      type="button"
                    >
                      <svg
                        class="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlink:href="#icon-bookmark"
                        />
                      </svg>
                      <span
                        class="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    class="place-card__rating rating"
                  >
                    <div
                      class="place-card__stars rating__stars"
                    >
                      <span
                        style="width: 80%;"
                      />
                      <span
                        class="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    class="place-card__name"
                  >
                    <a
                      href="/offer/1"
                    >
                      Penthouse, 4-5 rooms + 5 balconies
                    </a>
                  </h2>
                  <p
                    class="place-card__type"
                  >
                    hotel
                  </p>
                </div>
              </article>
            </div>
          </section>
          <div
            class="cities__right-section"
          >
            <section
              class="cities__map map leaflet-container leaflet-grab leaflet-touch-drag"
              style="position: relative;"
              tabindex="0"
            >
              <div
                class="leaflet-pane leaflet-map-pane"
                style="left: 0px; top: 0px;"
              >
                <div
                  class="leaflet-pane leaflet-tile-pane"
                >
                  <div
                    class="leaflet-layer "
                    style="z-index: 1;"
                  >
                    <div
                      class="leaflet-tile-container leaflet-zoom-animated"
                      style="z-index: 18; left: 0px; top: 0px;"
                    >
                      <img
                        alt=""
                        class="leaflet-tile"
                        role="presentation"
                        src="https://a.basemaps.cartocdn.com/rastertiles/voyager/13/4250/2734.png"
                        style="width: 256px; height: 256px; left: -51px; top: -84px;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="leaflet-pane leaflet-shadow-pane"
                />
                <div
                  class="leaflet-pane leaflet-overlay-pane"
                />
                <div
                  class="leaflet-pane leaflet-marker-pane"
                >
                  <img
                    alt=""
                    class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                    src="img/pin.svg"
                    style="margin-left: -20px; margin-top: -40px; width: 40px; height: 40px; left: 128px; top: 139px; z-index: 139;"
                    tabindex="0"
                  />
                </div>
                <div
                  class="leaflet-pane leaflet-tooltip-pane"
                />
                <div
                  class="leaflet-pane leaflet-popup-pane"
                />
              </div>
              <div
                class="leaflet-control-container"
              >
                <div
                  class="leaflet-top leaflet-left"
                >
                  <div
                    class="leaflet-control-zoom leaflet-bar leaflet-control"
                  >
                    <a
                      aria-label="Zoom in"
                      class="leaflet-control-zoom-in"
                      href="#"
                      role="button"
                      title="Zoom in"
                    >
                      +
                    </a>
                    <a
                      aria-label="Zoom out"
                      class="leaflet-control-zoom-out"
                      href="#"
                      role="button"
                      title="Zoom out"
                    >
                      −
                    </a>
                  </div>
                </div>
                <div
                  class="leaflet-top leaflet-right"
                />
                <div
                  class="leaflet-bottom leaflet-left"
                />
                <div
                  class="leaflet-bottom leaflet-right"
                >
                  <div
                    class="leaflet-control-attribution leaflet-control"
                  >
                    <a
                      href="https://leafletjs.com"
                      title="A JS library for interactive maps"
                    >
                      Leaflet
                    </a>
                     | © 
                    <a
                      href="https://www.openstreetmap.org/copyright"
                    >
                      OpenStreetMap
                    </a>
                     contributors © 
                    <a
                      href="https://carto.com/attributions"
                    >
                      CARTO
                    </a>
                  </div>
                </div>
              </div>
            </section>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
